plugins/scaffolder/src/components/MultistepJsonForm/MultistepJsonForm.tsx

```diff
@@ -59,13 +114,19 @@ export const MultistepJsonForm = ({
   widgets,
 }: Props) => {
   const [activeStep, setActiveStep] = useState(0);
+  const [reviewData, setReviewData] = useState({});
 
   const handleReset = () => {
     setActiveStep(0);
     onReset();
   };
-  const handleNext = () =>
+  const handleNext = () => {
     setActiveStep(Math.min(activeStep + 1, steps.length));
+
+    if (Math.min(activeStep + 1, steps.length) === steps.length) {
```

Comment:
What's the use case of only doing this if it's the last step? Can't we just do `metadata={getReviewData(formData), steps}` and not have the internal state?

Reply:
I`m sure your way is better... Changing...